controllers/auth.controller.js

```diff
@@ -0,0 +1,134 @@
+const { User } = require('../models');
+const statusCode = require('http-status');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const cryptoRandomString = require('crypto-random-string');
+const sendMail = require('../utils/sendMail');
+const config = require('../config/index');
+
+exports.register = async (req, res) => {
+    const { email, password, phoneNumber, fullName } = req.body;
+    try {
+        const userExists = await User.findOne({ email });
+        if (userExists) {
+            return res.status(statusCode.CONFLICT).json({
+                message: 'User already exists'
+            });
+        }
+        const hashedPassword = await bcrypt.hash(password, 10);
+        const generatedToken = cryptoRandomString({length: 15, type: 'url-safe'});
+        const user = new User({
+            email,
+            password: hashedPassword,
+            phoneNumber,
+            fullName,
+            verificationToken: generatedToken,
+        });
+        const savedUser = await user.save();
+        const verifyUrl = `http://localhost:${config.PORT}/api/v1/auth/verify-account/${savedUser.email}-${generatedToken}`;
```

Comment:
Get the URL host dynamically, don't type `localhost` manually.
Something like this:
```suggestion
        const verifyUrl = `http://ip/api/v1/auth/verify-account/${savedUser.email}-${generatedToken}`;
```